cirq-google/cirq_google/ops/internal_gate.py

```diff
@@ -78,3 +99,46 @@ def _value_equality_values_(self):
             self._num_qubits,
             frozenset(self.gate_args.items()) if hashable else self.gate_args,
         )
+
+
+def encode_function(
+    x: Union[Sequence[float], np.ndarray],
+    y: Sequence[float],
```

Comment:
Can we use the same type for `x` and `y`?

Reply:
nope, `x` could be a multidimensional array. `y` is always a vector

Reply:
multidimensional `x` is not supported now but will be

Reply:
done